### PR TITLE
Example re-frame-test tests

### PR DESCRIPTION
**Pull request by samroberton**

### CODE REVIEW

**Comment by samroberton** [comment]
Note that as discussed on Slack, the purpose of this PR is just to have somewhere to put the tests I've been working on.  We don't expect to merge this PR, because we don't necessarily want these changes to add to the complexity of the todomvc example within the re-frame repo.

File: examples/todomvc/test/todomvc/core_test.cljc [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+      (is (empty? @sorted-todos))
+      (is (empty? @todos))
+      (is (empty? @visible-todos))
+      (is (= 0 @completed-count))
+
+      (rf/dispatch [:add-todo "write first test"])
+      (rf-test/wait-for [:add-todo]
+        (is (= [{:id 1, :title "write first test", :done false}] @todos))
+
+        (rf/dispatch [:add-todo "write second teXt"])
```
Comment (danielcompton) [inline examples/todomvc/test/todomvc/core_test.cljc]: One thing that could be good to add would be a testing dispatching one event which async dispatches a second, and waiting for the second. But don't worry right now, as not sure where/how we'll merge these tests.

File: src/re_frame/interop.clj [hunk 2]
```diff
@@ -24,7 +24,8 @@
 (defonce ^:private executor (Executors/newSingleThreadExecutor))
 
 (defn next-tick [f]
-  (.execute executor f)
+  (let [bound-f (bound-fn [& args] (apply f args))]
```
Comment (danielcompton) [inline src/re_frame/interop.clj]: Have made a commit to re-frame to add this

File: examples/todomvc/test/todomvc/core_test.cljc [hunk 3]
```diff
@@ -0,0 +1,126 @@
+(ns todomvc.core-test
+  (:require #?(:cljs [cljs.test :refer-macros [deftest is]]
+               :clj  [clojure.test :refer [deftest is]])
```
Comment (samroberton) [inline examples/todomvc/test/todomvc/core_test.cljc]: Fwiw, from ClojureScript 1.9.198 onwards we can eliminate the reader conditional here and just use the `:clj` branch, and ClojureScript will figure it out.

File: examples/todomvc/project.clj [hunk 4]
```diff
@@ -1,8 +1,9 @@
 (defproject todomvc-re-frame "0.8.0"
-  :dependencies [[org.clojure/clojure        "1.8.0"]
+  :dependencies [[org.clojure/clojure        "1.9.0-alpha10"]
                  [org.clojure/clojurescript  "1.9.89"]
                  [reagent "0.6.0-rc"]
-                 [re-frame "0.8.0-alpha4"]
+                 [re-frame "0.8.0-SNAPSHOT"]
+                 [day8.re-frame/test "0.1.0-SNAPSHOT"]
```
Comment (vikeri) [inline examples/todomvc/project.clj]: Just to clarify, `day8.re-frame/test` doesn't exist on Clojars yet?

Comment (danielcompton) [inline examples/todomvc/project.clj]: Yep it's up there, 0.1.3 is the latest version: https://clojars.org/day8.re-frame/test

**Comment by mike-thompson-day8** [comment]
Closing.  All parts of this example have been transferred to re-frame-test repo.

**Pull request closed without merging**